Makefile

```diff
@@ -162,26 +162,25 @@ check-tidy:
 
 .PHONY: build-image
 build-image:
-	# Add --pull so that we are using the latest base image.
-	docker build --pull --file ./cmd/$(TARGET)/Dockerfile --tag $(IMAGE_NAME) --build-arg GIT_HASH=$(GIT_HASH) .
-
-.PHONY: tag-image
-tag-image: DOCKER_IMAGE ::= quay.io/theauthgear/$(IMAGE_NAME)
-tag-image:
-	docker tag $(IMAGE_NAME) $(DOCKER_IMAGE):latest
-	docker tag $(IMAGE_NAME) $(DOCKER_IMAGE):$(GIT_HASH)
-	if [ ! -z $(GIT_NAME) ]; then docker tag $(IMAGE_NAME) $(DOCKER_IMAGE):$(GIT_NAME); fi
-
-.PHONY: push-image
-push-image: DOCKER_IMAGE ::= quay.io/theauthgear/$(IMAGE_NAME)
-push-image:
-	docker manifest inspect $(DOCKER_IMAGE):$(GIT_HASH) > /dev/null; if [ $$? -eq 0 ]; then \
-		echo "$(DOCKER_IMAGE):$(GIT_HASH) exists. Skip push"; \
-	else \
-		docker push $(DOCKER_IMAGE):latest ;\
-		docker push $(DOCKER_IMAGE):$(GIT_HASH) ;\
-		if [ ! -z $(GIT_NAME) ]; then docker push $(DOCKER_IMAGE):$(GIT_NAME); fi ;\
-	fi
+	$(eval DOCKER_IMAGE ::= quay.io/theauthgear/$(IMAGE_NAME))
+	$(eval BUILD_OPTS ::= --tag $(DOCKER_IMAGE))
```

Comment:
To declare target-specific variable, follow this syntax https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html

```makefile
build-image: DOCKER_IMAGE ::= quay.io/theauthgear/$(IMAGE_NAME)
build-image: BUILD_OPTS ::= --tag $(DOCKER_IMAGE)
build-image:
  # The original body of this target goes here.
```

Reply:
This comment is outdated.